File: content/courses/parallel-computing-introduction/codes/omp_antidep.f90
program antidep

    integer, parameter    :: N=48
    integer, dimension(N) :: a
    integer               :: i

    do i=1,N
       a(i)=100-i
    enddo

    !$omp parallel do
    do i=1, N-1
        a(i)=a(i+1)+10
    enddo
    !$omp end parallel do

    do i=1, N
       write(*,*) i,a(i)
    enddo

end program

File: content/courses/parallel-computing-introduction/codes/omp_antidep.c
#include <stdio.h>
#include <omp.h>

int main() {

    const int N=48;
    int a[N];

    for (int i=0; i<N; i++) 
        a[i]=100-i;

#pragma omp parallel for 
    for (int i=0; i<N-1; i++) {
        a[i] = a[i+1] + 10;
    }

    for (int i=0; i<N; i++) {
        printf("%d %d\n",i,a[i]);
    }

}
